File: go/gildedrose/backstage_pass_item.go
package gildedrose

type BackstagePass struct {
	*Item
}

func (item *BackstagePass) itemUpdateStrategy() {
	item.incQuality()
	if item.SellIn <= 10 {
		item.incQuality()
	}

	if item.SellIn <= 5 {
		item.incQuality()
	}
	item.SellIn--
	if item.SellIn < 0 {
		item.Quality = 0
	}
}

File: Java/src/main/java/com/gildedrose/Items/Concrete/BackstagePassItem.java
package com.gildedrose.Items.Concrete;

import com.gildedrose.Item;
import com.gildedrose.Items.GildedItem;

public class BackstagePassItem extends GildedItem {
    public BackstagePassItem(Item item) {
        super(item);
    }

    @Override
    public void updateItemQuality() {
        incrementItemQuality();
        if (item.sellIn < 11) {
            incrementItemQuality();
        }

        if (item.sellIn < 6) {
            incrementItemQuality();
        }

        item.sellIn--;

        if (item.sellIn < 0) {
            item.quality = 0;
        }
    }
}
